File: packages/schema/src/user/staff-member.ts
/**
 * Staff Member Schema Definitions
 *
 * IMPORTANT: Follow conventions in utils.ts header comment
 * - Use snake_case for all field names
 * - Use .nullable() for Option<T> fields, not .optional()
 */

import { z } from "zod";
import { idSchema, dateTimeSchema } from "../utils";

export const staffMemberSchema = z.object({
  id: idSchema,
  employee_id: z.string(),
  first_name: z.string(),
  last_name: z.string(),
  email: z.string(),
  position: z.string(),
  department: z.string().nullable(),
  branch_id: idSchema.nullable(),
  hire_date: dateTimeSchema,
  is_active: z.boolean(),
  created_at: dateTimeSchema,
  updated_at: dateTimeSchema,
});

export type StaffMember = z.infer<typeof staffMemberSchema>;

File: apps/desktop/crates/infrastructure/src/user/entities/staff_member.rs
use pharos_core::{DateTime, Id};
use sea_orm::entity::prelude::*;
use serde::{Deserialize, Serialize};

/// Staff member (employee) information
#[derive(Clone, Debug, PartialEq, Eq, DeriveEntityModel, Serialize, Deserialize)]
#[sea_orm(table_name = "staff_members")]
pub struct Model {
    #[sea_orm(primary_key, auto_increment = false)]
    pub id: Id,

    #[sea_orm(unique)]
    pub employee_id: String,

    pub first_name: String,
    pub last_name: String,

    #[sea_orm(unique)]
    pub email: String,

    pub position: String,

    #[sea_orm(nullable)]
    pub department: Option<String>,

    #[sea_orm(nullable)]
    pub branch_id: Option<Id>,

    pub hire_date: DateTime,
    pub is_active: bool,
    pub created_at: DateTime,
    pub updated_at: DateTime,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {
    #[sea_orm(
        belongs_to = "crate::branch::entities::branch::Entity",
        from = "Column::BranchId",
        to = "crate::branch::entities::branch::Column::Id",
        on_delete = "SetNull"
    )]
    Branch,
}

#[async_trait::async_trait]
impl ActiveModelBehavior for ActiveModel {
    fn new() -> Self {
        Self {
            id: sea_orm::ActiveValue::Set(Id::new()),
            is_active: sea_orm::ActiveValue::Set(true),
            created_at: sea_orm::ActiveValue::Set(DateTime::now()),
            updated_at: sea_orm::ActiveValue::Set(DateTime::now()),
            ..Default::default()
        }
    }

    async fn before_save<C>(mut self, _db: &C, insert: bool) -> Result<Self, DbErr>
    where
        C: ConnectionTrait,
    {
        if !insert {
            self.updated_at = sea_orm::ActiveValue::Set(DateTime::now());
        }
        Ok(self)
    }
}
